src/test/kotlin/minesweeper/domain/board/MineBoardTest.kt

```diff
@@ -0,0 +1,74 @@
+package minesweeper.domain.board
+
+import minesweeper.domain.cell.Empty
+import minesweeper.domain.cell.Mine
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.Assertions.assertThatThrownBy
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.Arguments
+import org.junit.jupiter.params.provider.MethodSource
+import java.util.stream.Stream
+
+internal class MineBoardTest {
+
+    @ParameterizedTest
+    @MethodSource("지뢰 보드의 높이와 너비 중 음수가 존재하는 케이스")
+    fun `지뢰 보드의 높이와 너비는 음수일 수 없다`(width: Int, height: Int) {
+        assertThatThrownBy { newMineBoard(width, height, DEFAULT_NUMBER_OF_MINES) }
```

Comment:
```suggestion
        assertThrows<IllegalArgumentException> { newMineBoard(width, height, DEFAULT_NUMBER_OF_MINES) }
```
저는 assertThatThrownBy 보다 assertThrows을 선호하고 있어요.

assertThatThrownBy와 assertThrows의 차이가 뭘까요?

Reply:
두 함수는 사용되는 케이스는 동일하지만 제공하는 패키지가 다르다는 차이점이 있는 것 같은데요~

이 부분은 제가 tc를 작성할 때 마다 항상 assertj 패키지를 사용하다보니.. 자연스럽게 사용하게 된 것 같습니다 ㅎㅎ

개인적으로는 assertj 에서 제공하는 assertThatThrownBy 는 체이닝으로 예외 타입과 예외 메시지를 확인하는 코드를 가독성 좋게 작성할 수 있어서 사용했습니다.

혹시 형준님께서 assertThrows를 선호하시는 이유를 말씀해주실 수 있으신가요?? (궁금)

Reply:
assertThatThrownBy()과 assertThrows<>의 가장 큰 차이점은 .class.java 파일을 쓸 것인지 .kt 파일을 쓸 것인지에 대한 차이점이 있습니다. 이러한 차이점이 있기 때문에 assertThatThrownBy은 java Exception class 만 받을 수 있습니다. 코틀린 Exception class를 받기 위해서도 결국 java로 변경해서 사용을 해야될거에요.하지만, assertThrows<>는 이미 .kt 파일로 만들어졌기 때문에, kotlin의 익셉션 처리를 할 수 있고, 순수한 kotiln API로 에러 테스트를 할 수 있게 만듭니다.

저의 개인적인 생각이지만, kotlin 코틀린스럽게 짜는 코드는 kotlin 프로젝트에서 자바와 혼용해서 짜는 것이 아닌, kotlin에 있는 API만으로 코드를 짜는 것부터 시작한다고 생각합니다 : )

앞으로 있을 미션에 대해 java API가 아닌, kotlin API로 짜는 연습을 해보시길 추천드립니다! 😁